lang/LangPrimSource/SC_Ableton_Link.cpp

```diff
@@ -37,30 +22,39 @@ class LinkClock : public TempoClock
 	void SetTempoAtBeat(double inTempo, double inBeats);
 	void SetTempoAtTime(double inTempo, double inSeconds);
 	void SetAll(double inTempo, double inBeats, double inSeconds);
-	double BeatsToSecs(double beats) const
-	{
-		auto sessionState = mLink.captureAppSessionState();
-		double secs = linkToHrTime(sessionState.timeAtBeat(beats, mQuantum)) - mLatency;
-		return secs;
-	}
-	double SecsToBeats(double secs) const
-	{
-		auto sessionState = mLink.captureAppSessionState();
-		double beats = sessionState.beatAtTime(hrToLinkTime(secs + mLatency), mQuantum);
-		return beats;
-	}
+	double BeatsToSecs(double beats) const;
+	double SecsToBeats(double secs) const;
 
 	void SetQuantum(double quantum);
         double GetLatency();
         void SetLatency(double latency);
 	std::size_t NumPeers() const { return mLink.numPeers(); }
 
+  static void Init();
+  static std::chrono::microseconds GetInitTime() { return LinkClock::mInitTime; }
+
 private:
 	ableton::Link mLink;
 	double mQuantum;
 	double mLatency;
+  static std::chrono::microseconds mInitTime;
 };
 
+std::chrono::microseconds LinkClock::mInitTime;
```

Comment:
An extra line is required to create the linker definition of `std::chrono::microseconds LinkClock::mInitTime`.

This could have an initial value that `LinkClock::Init` checks against to avoid overwriting on double-calls to `LinkClock::Init`.  This is baked-into SuperCollider, so unlikely.

Reply:
> This could have an initial value that `LinkClock::Init` checks against to avoid overwriting on double-calls to `LinkClock::Init`. This is baked-into SuperCollider, so unlikely.

I don't think there's a risk of that happening. Besides, there is no restriction AFAICT that prevents someone from using the headers for libsclang in such a way that the VM runtime is created and destroyed several times over the life of the program.